docs/explain/POST.md

```diff
@@ -0,0 +1,72 @@
+Implementing a POST endpoint
+=============================
+
+# Setup
+
+This assumes you have a Rails app with a model `Cheese` defined, mapping to a database (e.g. SQLite). 
+It should have one required column, `name`.
```

Comment:
this does a better job explaining how the API & models relate. I feel we need a section that explains how we can support this though as often you want an API attribute to have a different name to whats in the database (especially for Public APIs). We should have a section in these how tos for how to have an API value populated by calling a rails service (doesn't have to be as part of this PR though)

Reply:
Since you have some experience in that regard, maybe you want to add such a section to this PR?

Reply:
``` ruby
# In order to add new fields to an API the easiest option is to create an attr_reader on the model you're referencing. For instance app/controllers/api/v3/api_builder_definitions/shared/document.rb has the 4 fields: 
[:file_name, :file_size, :created_at, :content_type]. 
# These fields can be whatever you want to use for the api and don't tie to anything in the App itself.
# 
# The "source" is where we pull in data for these 4: 
[:document_file_name, :document_content_type, :document_file_size, :document_created_at].
# These four "sources" are defined in app/models/document.rb as attr_reader methods with the same names as the source. 

# While its not ideal, if the value you need is not stored ready to go in the database, your attr_reader on the model COULD make a call to a service in order to return the value desired but this isn't really a desirable approach as it shouldn't be the concern of the model. Additional investigation is needed into alternative ways of managing this.
```

Reply:
thoughts?

Reply:
Sounds – unfortunately – about right. It's cumbersome, to put it short. Anyhow, should we put this in the how-tos in this PR?

Reply:
There's 3 ways of doing this?

`attr_reader` shouldn't be required for the simple situation @CitationNerded notes above…seems excessive for 2/4 of them…but maybe it's also a preferred way to do logic beyond a 1:1 mapping (`document_file_name`, etc).

1. 1:1 naming (source === name)
2. via `source` field
3. via a block|lambda|proc|whatever it's called here

```ruby
# nested + boolean cast:
field(name: :supported, type: BOOLEAN, doc: 'Whether or not Sharesight fully supports the nominal country') { |o| !!o.nominal_country&.ss_support? }

# re-map (these are equal):
field(:financial_year_end, doc: 'The end of the financial year (MM-DD)') { |p| p.financial_year_end_s }
field(:financial_year_end, source: :financial_year_end_s, doc: 'The end of the financial year (MM-DD)')
```

Reply:
i have not tried using procs and lambdas to set these fields so haven't written much additional in the documentation around it beyond the examples you give here @kylorhall  but did add a section for it and the other two options near the bottom of README.md